packages/vscode-docker-registries/src/auth/DockerHubAuthProvider.ts

```diff
@@ -0,0 +1,62 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { AuthenticationProvider } from '../contracts/AuthenticationProvider';
+import { LoginInformation } from '../contracts/BasicCredentials';
+import { DockerHubRegistryUrl, DockerHubRequestUrl } from '../clients/DockerHub/DockerHubRegistryDataProvider';
+import { httpRequest } from '../utils/httpRequest';
+import { BasicAuthProvider } from './BasicAuthProvider';
+
+export class DockerHubAuthProvider extends BasicAuthProvider implements AuthenticationProvider {
+    // TODO: this token expires after a month, should we bother to actually refresh it?

```

Comment:
Do?

Reply:
Nahh, who keeps VSCode running for a month?